packages/storage/src/persisted.ts

```diff
@@ -152,33 +173,73 @@ export function makePersisted<T, O extends Record<string, any> = {}>(
   const name = options.name || `storage-${createUniqueId()}`;
   const serialize: (data: T) => string = options.serialize || JSON.stringify.bind(JSON);
   const deserialize: (data: string) => T = options.deserialize || JSON.parse.bind(JSON);
-  const init = storage.getItem(name, storageOptions);
   const set =
     typeof signal[0] === "function"
       ? (data: string) => (signal[1] as any)(() => deserialize(data))
       : (data: string) => (signal[1] as any)(reconcile(deserialize(data)));
   let unchanged = true;
 
-  if (init instanceof Promise) init.then(data => unchanged && data && set(data));
-  else if (init) set(init);
+  function fetchFromStorage() {
+    const value = storage.getItem(name, storageOptions);
+    if (value)
+      if (typeof value === "object" && "then" in value) // Check if value is a promise
+        value.then(data => unchanged && data && set(data));
+      else set(value);
+  }
+
+  fetchFromStorage() // Initialize the underlying signal
+
+  let broadcastChannel: BroadcastChannel | undefined
```

Comment:
I'd prefer a mechanism that allows to define your own IPC, with BroadcastChannel as a fallback.

Reply:
i don't really see the use case, because if you want to handle it your self you can just wrap the signal with `sync: false` and handle the sync for your self. BroadcastChannels are the officially preferred way of inter window and tab communication so i also don't see why we should give first party support for something like `localstorage` change events.

Reply:
But otherwise, we carry useless code that is difficult to tree-shake, and in most cases, you can get by without sync. So IMO it would be better to have an API and one BroadcastChannel-based plugin to use it.